WidgetKit/TopSites/TopSitesWidget.swift

```diff
@@ -0,0 +1,136 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import SwiftUI
+import WidgetKit
+import Combine
+import Shared
+import Storage
+import SyncTelemetry
+
+struct TopSitesProvider: TimelineProvider {
+    public typealias Entry = TopSitesEntry
+
+    public func snapshot(with context: Context, completion: @escaping (TopSitesEntry) -> ()) {
+        TopSitesHandler.getTopSites(profile: BrowserProfile(localName: "profile")).uponQueue(.main) { topSites in
+            
+            // TODO: It seems like some sites (Twitter, YouTube) do not have favicons... where should I fetch them? I think they're "built in"
```

Comment:
I believe we have local fevicons that we can fetch and incase they are outdated we should update them in our app.